src/main/java/lotto/domain/Lotto.java

```diff
@@ -0,0 +1,41 @@
+package lotto.domain;
+
+import java.util.Collections;
+import java.util.List;
+
+public class Lotto {
+    public static final int COUNT = 6;
+    private final List<Integer> numbers;
+
+    public Lotto(NumberGeneration numberGeneration) {
+        this.numbers = makeLotto(numberGeneration);
+    }
+
+    public static List<Integer> makeLotto(NumberGeneration numberGeneration) {
+        return numberGeneration.generate();
+    }
+
+    public int countMatchNumber(List<Integer> winningList) {
+        int count = 0;
+        for (int i = 0; i < COUNT; i++) {
+            count += checkMatch(winningList, numbers.get(i));
+        }
+        return count;
+    }
+
+    private int checkMatch(List<Integer> winningList, int matchNumber) {
+        if (contains(winningList, matchNumber)) {
+            return 1;
+        }
+        return 0;
+    }
+
+    private boolean contains(List<Integer> winningList, int matchNumber) {
+        return winningList.contains(matchNumber);
+    }
```

Comment:
```suggestion
    private boolean contains(int matchNumber) {
        return numbers.contains(matchNumber);
    }
```

내부상태가아닌 전달받게끔 구현한 이유가 궁금해요!

Reply:
countMatchNumber 메서드 내부 for문은 로또 객체를 기준으로 돌아갑니다. 자동으로 생성된 로또 숫자를 하나씩 뽑아서 해당 번호가 당첨 번호 전체에 포함 되어 있는지 확인하는 방식으로 구현하였습니다. 

다시 생각해보니 for문 기준을 당첨로또번호로 두고,  로또 객체의 numbers에 당첨 번호가 포함되어 있는지 확인하는 방법으로 구현하는 것이 더 직관적일 것 같습니다! 매개변수도 줄일 수 있구요

피드백 반영하여 수정하겠습니다!